install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,476 @@ toc::[]
+of the various fields. In particular, *.State* has the set of admitted routes.
+
+.Router Configuration Parameters
+[cols="3*", options="header"]
+|===
+|Field | Type | Description
+|`*WorkingDir*` | string  | the directory that files will be written to,
+defaults to /var/lib/containers/router
+|`*State*` | map[string](ServiceAliasConfig) | the routes
+|`*ServiceUnits*` | map[string]ServiceUnit | the service lookup
+|`*DefaultCertificate*` | string | full path and file name to the default
+certificate in pem format.
+|`*PeerEndpoints*` | []Endpoint | peers
+|`*StatsUser*` | string | username to expose stats with (if the template
+supports it)
+|`*StatsPassword*` | string | password to expose stats with (if the template
+supports it)
+|`*StatsPort*` | int | port to expose stats with (if the template supports it)
+|`*BindPorts*` | bool | whether the router should bind the default ports
+|===
+
+.Router ServiceAliasConfig (A Route)
+[cols="3*", options="header"]
+|===
+|Field | Type | Description
+|`*Name*` | string | Name is the user-specified name of the route.
+|`*Namespace*`| string | Namespace is the namespace of the route.
+|`*Host*` | string | Host is the host name ie. www.example.com
+|`*Path*` | string | Path is an optional path ie. www.example.com/myservice
+where "myservice" is the path
+|`*TLSTermination*` | routeapi.TLSTerminationType | TLSTermination is the
+termination policy for this backend and drives the mapping files and router
+configuration
+|`*Certificates*` | map[string]Certificate | Certificates used for securing
+this backend.  Keyed by the cert id
+|`*Status*` | ServiceAliasConfigStatus | Indicates the status of configuration
+that needs to be persisted.  Right now this only includes the certificates and
+is not an indicator of being written to the underlying router implementation
+|`*PreferPort*` | string | Indicates the port the user wishes to expose. If
+empty, a port will be selected for the service.
+|`*InsecureEdgeTerminationPolicy*` | routeapi.InsecureEdgeTerminationPolicyType |
+InsecureEdgeTerminationPolicy indicates desired behavior for insecure
+connections to an edge-terminated route: none (or disable), allow or redirect
+|`*RoutingKeyName*` | string | Hash of the route+namespace name - used to
+obscure cookieId
+|`*IsWildcard*` | bool | IsWildcard indicates this service unit needs
+wildcarding support.
+|`*Annotations*` | map[string]string | Annotations attached to this route
+|`*ServiceUnitNames*` | map[string]int32 | ServiceUnitNames is a collection
+of services that support this route, keyed by service name and valued on the
+weight attached to it with respect to other entries in the map
+|`*ActiveServiceUnits*` | int | ActiveServiceUnits is a count of the
+ServiceUnitNames with a non-zero weight
+|===
+
+The ServiceAliasConfig is a route for a service.  Uniquely identified by
+host + path.  The template usually uses $cfg.Field to access the fields of
```

Comment:
When I put the discussion before the table, the table numbering is reset. How do you code this?

Added new wording.